core/testdata/config-full.toml

```diff
@@ -0,0 +1,259 @@
+IncomingMessageBufferSize = 3
+DHTLookupInterval = 9
+BootstrapCheckInterval = '1m0s'
+NewStreamTimeout = '1s'
+
+[P2P]
+NetworkingStack = 'V1V2'
+IncomingMessageBufferSize = 13
+OutgoingMessageBufferSize = 17
+AnnouncePort = 1234
+BootstrapCheckInterval = '1m0s'
+BootstrapPeers = ['foo', 'bar', 'should', 'these', 'be', 'typed']
+DHTAnnouncementCounterUserPrefix = 4321
+DHTLookupInterval = 9
+ListenPort = 9
+NewStreamTimeout = '1s'
+PeerID = '12D3KooWMoejJznyDuEk5aX6GvbjaG12UzeornPCBNzMRqdwrFJw'
+PeerstoreWriteInterval = '1m0s'
+V2AnnounceAddresses = ['a', 'b', 'c']
+V2Bootstrappers = ['1', '2', '3']
+V2DeltaDial = '1m0s'
+V2DeltaReconcile = '1s'
+V2ListenAddresses = ['foo', 'bar']
+
+[Keeper]
+CheckUpkeepGasPriceFeatureEnabled = true
+DefaultTransactionQueueDepth = 17
+GasPriceBufferPercent = 12
+GasTipCapBufferPercent = 43
+BaseFeeBufferPercent = 89
+MaximumGracePeriod = 31
+RegistryCheckGasOverhead = 90
+RegistryPerformGasOverhead = 18446744073709551615
+RegistrySyncInterval = '1h0m0s'
+RegistrySyncUpkeepQueueSize = 31
+TurnLookBack = 91
+TurnFlagEnabled = true
+
+[AutoPprof]
+Enabled = true
+ProfileRoot = 'prof/root'
+PollInterval = '1m0s'
+GatherDuration = '12s'
+GatherTraceDuration = '13s'
+MaxProfileSize = '1.00gb'
+CPUProfileRate = 7
+MemProfileRate = 9
+BlockProfileRate = 5
+MutexProfileFraction = 2
+MemThreshold = '1.00gb'
+GoroutineThreshold = 999
+
+[EVM]
+
+[EVM.1]
+BalanceMonitorEnabled = true
+BlockBackfillDepth = 100
+BlockBackfillSkip = true
+BlockEmissionIdleWarningThreshold = '1h0m0s'
+ChainType = 'Optimism'
+EIP1559DynamicFees = true
+FinalityDepth = 42
+FlagsContractAddress = '0xabcd1234'
+GasBumpPercent = 10
+GasBumpTxDepth = 6
+GasBumpWei = '100'
+GasEstimatorMode = 'L2Suggested'
+GasFeeCapDefault = '9223372036854775807'
+GasLimitDefault = 12
+GasLimitMultiplier = 7.0
+GasPriceDefault = '9223372036854775807'
+GasTipCapDefault = '2'
+GasTipCapMinimum = '1'
+HeadTrackerHistoryDepth = 15
+HeadTrackerMaxBufferSize = 17
+HeadTrackerSamplingInterval = '1h0m0s'
+LinkContractAddress = '0x1234abcd'
+LogBackfillBatchSize = 17
+LogPollInterval = '1m0s'
+MaxGasPriceWei = '281474976710655'
+MaxInFlightTransactions = 19
+MaxQueuedTransactions = 99
+MinGasPriceWei = '13'
+MinIncomingConfirmations = 13
+MinimumContractPayment = '9223372036854775807'
+NodeNoNewHeadsThreshold = '1m0s'
+NodePollFailureThreshold = 5
+NodePollInterval = '1m0s'
+NonceAutoSync = true
+OCRContractConfirmations = 11
+OCRContractTransmitterTransmitTimeout = '1m0s'
+OCRDatabaseTimeout = '1s'
+OCRObservationGracePeriod = '1s'
+OCR2ContractConfirmations = 7
+OperatorFactoryAddress = '0x9876qwerty'
+TxReaperInterval = '1m0s'
+TxReaperThreshold = '1m0s'
+TxResendAfterThreshold = '1h0m0s'
+UseForwarders = true
+
+[EVM.1.BlockHistoryEstimator]
+BatchSize = 17
+BlockDelay = 10
+BlockHistorySize = 12
+EIP1559FeeCapBufferBlocks = 13
+TransactionPercentile = 15
+
+[EVM.1.KeySpecific]
+
+[EVM.1.KeySpecific.0x1234]
+MaxGasPriceWei = '79228162514264337593543950335'
```

Comment:
We'll need to decide how to handle large (outside int64 range) integers. We can use native TOML numbers (drop the quotes) with custom types pretty easily, but the spec only requires decoders to handle up to int64, so we would risk being incompatible with other TOML tooling. We would also have to trust that other tools properly reject integers they cannot handle, instead of e.g. falling back to a lossy value and mangling things.

Reply:
I'm leaning towards prioritizing nodeop UX by using regular, unquoted values. My only concerns are the UI, or if we end up having to parse arbitrary toml without the strict custom types :thinking:

Reply:
I was mistaken about our ability to customize here. Custom types with v2 of this toml library are actually restricted to emitting strings via `MarshalText()` that will be quoted :disappointed: